gazebo_plugins/src/gazebo_ros_diff_drive.cpp

```diff
@@ -230,41 +233,60 @@ void GazeboRosDiffDrive::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr
   // Initialize ROS node
   impl_->ros_node_ = gazebo_ros::Node::Get(_sdf);
 
-  // Get joints
-  impl_->joints_.resize(2);
-
-  auto left_joint = _sdf->Get<std::string>("left_joint", "left_joint").first;
```

Comment:
This change breaks backwards compatibility with previous versions of the plugin, so all existing models that use `left_joint` wouldn't work anymore.

How about instead of using `num_wheel_pairs` + `left_jointN`, we use multiple `left_joint`. The code could iterate over them doing something like this:

```.cpp
  for (auto left_joint_elem = sdf->GetElement("left_joint"); 
        left_joint_elem != nullptr; 
        left_joint_elem = left_joint_elem->GetNextElement("left_joint") ) {
    // Store value
  }
```

And then sanity check that there's the same number of left and right joints.

The same goes for `wheel_separation`, etc.